OTGTool/src/main/scala/t/db/file/TSVMetadata.scala

```diff
@@ -57,12 +56,17 @@ object TSVMetadata {
         (attrib, values) <- columns
         normalizedID = normalizeAttributeID(attrib)
         trimmed = values.map(_.trim)
-        attribute = attrSet.byIdLowercase.getOrElse(normalizedID,
-            {
+        attribute = {
               val attribType = getAttribType(attrib, headerRows)
-              warningHandler(s"attribute $normalizedID not found, creating new as type $attribType")
-              attrSet.findOrCreate(normalizedID, attrib, attribType.toLowerCase)
-            })
+              if (attrSet.mayRedefine(normalizedID) && attrSet.hasId(normalizedID)) {
+                warningHandler(s"Redefining attribute $normalizedID with type $attribType")
+              } else if (!attrSet.hasId(normalizedID)) {
+                warningHandler(s"attribute $normalizedID not found, creating new as type $attribType")
+              } else {
+                warningHandler(s"Attribute $normalizedID exists and may not be redefined")
```

Comment:
Yes, but then I would have to create additional logic to really identify the illegal circumstances, since specifying e.g. sample_id and platform is part of normal data insertion. For now, I will change the message to sound less threatening instead since this is not really something wrong that the user is doing.

Reply:
Right, yes, I think I misunderstood what those cases meant, thanks for the change!

I think my confusion partly stems from it being called a `warningHandler`

Reply:
It's indeed misleading. Let's fix that too.